sunpy/instr/rhessi.py

```diff
@@ -348,3 +400,185 @@ def _build_energy_bands(label, bands):
+        Specifies by how much time the single files are separated.
+        Defaults to ``relativedelta(months=+1)``.
+
+    Returns
+    -------
+    ``pandas.DataFrame``
+        out : ``pandas.DataFrame`` containing the flares within the given time constraints
+    """
+
+    formats = {
+        5: "%y-%m",  # YY-mm
+        6: "%Y%m",  # YYYYmm
+        7: "%Y-%m",  # YYYY-mm
+        8: "%Y%m%d",  # YYYYmmdd
+        10: "%Y-%m-%d",  # YYYY-mm-dd
+        19: "%Y-%m-%dT%H:%M:%S",  # YYYY-mm-ddThh:MM:ss
+    }
+    try:
+        start_dt = datetime.strptime(start, formats[len(start)])
+        end_dt = datetime.strptime(end, formats[len(end)])
+    except (KeyError, ValueError):
+        raise ValueError("invalid datetime")
+
+    format_str = file_format[file_format.index("%"):file_format.rindex("%") + 2]
+    cur_format = start_dt.strftime(format_str)
+    end_format = end_dt.strftime(format_str)
+
+    if source in known_flare_list_sources:
+        source = known_flare_list_sources[source]
+
+    cur_dt = start_dt
+    result = pd.DataFrame()
+    while cur_format <= end_format:
+        file = file_format.replace(format_str, cur_format)
+        result = result.append(read_flare_file(source + file), ignore_index=True)
+        cur_dt = cur_dt + inc
+        cur_format = cur_dt.strftime(format_str)
+
+    # filter results for more detailed time constraints (if applicable)
+    if len(end) < 8:
+        end_dt += relativedelta(months=+1)  # add month if no further constraints were defined
+    elif len(end) <= 10:
+        end_dt += relativedelta(days=+1)  # add day if end date was specified on a day-basis
+    else:
+        end_dt += relativedelta(microsecond=+1)  # add 1ms so "smaller" operator works as intended
+
+    result = result[result['END_TIME'] >= start_dt]
+    result = result[result['START_TIME'] < end_dt]
+    return result
+
+
+def read_flare_file(file):
+    """
+    Read RHESSI flare list .fits file into ``pandas.DataFrame``. TIME values are parsed with
+    format 'utime', which is the same as Unix timestamp but starts 9 years later.
+    FLAGS are assigned their respective label (FLAG ID) and returned as `dict`.
+
+    Parameters
+    ----------
+    file : `str`
+        The URL or local filename of the hessi flare list.
+
+    Returns
+    -------
+    ``pandas.DataFrame``
+        out : ``pandas.DataFrame`` containing the parsed flares.
+
+    Examples
+    --------
+    >>> from sunpy.instr.rhessi as read_flare_file as rff
+    >>> rff("https://hesperia.gsfc.nasa.gov/hessidata/dbase/hessi_flare_list_201802.fits")
+
+    References
+    ----------
+    https://hesperia.gsfc.nasa.gov/rhessi3/data-access/rhessi-data/flare-list/index.html
+    """
+    try:
+        fits = sunpy.io.fits.read(file)
+    except Exception:
+        raise RuntimeError("couldn't load file " + file)
+
+    results = []
+    for row in fits[3].data[0:20]:
+        result_row = {}
+        for k in fits_rec_keys:
```

Comment:
I'm probably missing something but could you do `for k in fits[3].data.columns.names` and remove the constant at the top of the file or is than an optimisation?

Reply:
Nope, you're absolutely right :) I simply didn't know this exists..